tests/test_volume.py

```diff
@@ -261,7 +284,44 @@ def test_save_load(vols_1):
         assert np.allclose(vols_1, vols_loaded_double)
 
 
-def test_project(vols_1, dtype):
+def test_project(vols_hot_cold):
+    """
+    We project Volumes containing random hot/cold spots using random rotations and check that
+    hot/cold spots in the projections are in the expected locations.
+    """
+    vols, hot_cold_locs, vol_center = vols_hot_cold
+    dtype = vols.dtype
+    L = vols.resolution
+
+    # Generate random rotations.
+    rots = Rotation.generate_random_rotations(n=vols.n_vols, dtype=dtype)
+
+    # To find the expected location of hot/cold spots in the projections we rotate the 3D
+    # vector of locations by the transpose, ie. rots.invert(), (since our projections are
+    # produced by rotating the underlying grid) and then project along the z-axis.
+
+    # Expected location of hot/cold spots relative to (0, 0, 0) origin in (x, y, z) order.
+    expected_hot_cold = np.transpose(
```

Comment:
to be easier, i might have just done `pts_T = ....` , `rots_T= `, ... 

maybe you can do hot_cold_locs @ rots.matrices  and add a comment 
 about transposingt he data along the stack.  :innocent:
 
If none of that is actually better when its written out,  this is fine though.

Reply:
Much cleaner. Thanks! Done